pkg/apihelper/k8shelpers.go

```diff
@@ -26,50 +26,23 @@ import (
 	"k8s.io/apimachinery/pkg/types"
 	k8sclient "k8s.io/client-go/kubernetes"
 	restclient "k8s.io/client-go/rest"
-	"k8s.io/client-go/tools/clientcmd"
 )
 
 // Implements APIHelpers
 type K8sHelpers struct {
-	Kubeconfig string
+	Kubeconfig *restclient.Config
 }
 
 func (h K8sHelpers) GetClient() (*k8sclient.Clientset, error) {
-	// Set up an in-cluster K8S client.
-	var config *restclient.Config
-	var err error
-
-	if h.Kubeconfig == "" {
-		config, err = restclient.InClusterConfig()
-	} else {
-		config, err = clientcmd.BuildConfigFromFlags("", h.Kubeconfig)
-	}
-	if err != nil {
-		return nil, err
-	}
-
-	clientset, err := k8sclient.NewForConfig(config)
+	clientset, err := k8sclient.NewForConfig(h.Kubeconfig)
 	if err != nil {
 		return nil, err
 	}
 	return clientset, nil
 }
 
 func (h K8sHelpers) GetTopologyClient() (*topologyclientset.Clientset, error) {
-	// Set up an in-cluster K8S client.
-	var config *restclient.Config
-	var err error
-
-	if h.Kubeconfig == "" {
-		config, err = restclient.InClusterConfig()
-	} else {
-		config, err = clientcmd.BuildConfigFromFlags("", h.Kubeconfig)
-	}
-	if err != nil {
-		return nil, err
-	}
-
-	topologyClient, err := topologyclientset.NewForConfig(config)
+	topologyClient, err := topologyclientset.NewForConfig(h.Kubeconfig)
```

Comment:
Is this intended?

Reply:
Yes, it was intended. But good that you pointed out because I realized that the commit doing this could be improved, sharing more code.

I now updated the PR, hopefully making it better, creating a generic `GetKubeconfig` that is used by both nfd-master and nfd-topology-updater